instrumentation/resque/lib/opentelemetry/instrumentation/resque/patches/resque_module.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      module Patches
+        # Module to prepend to Resque for instrumentation
+        module ResqueModule
+          def self.prepended(base)
+            class << base
+              prepend ClassMethods
+            end
+          end
+
+          # Module to prepend to Resque singleton class
+          module ClassMethods
+            def push(queue, item)
+              attributes = {
+                'messaging.system' => 'resque',
+                'messaging.destination' => queue.to_s,
+                'messaging.destination_kind' => 'queue'
+              }
+
+              if (job_class = item[:class])
+                attributes['messaging.resque.job_class'] = job_class
+              end
+
+              tracer.in_span("#{queue} send", attributes: attributes, kind: :producer) do
+                OpenTelemetry.propagation.inject(item)
```

Comment:
Also similar to comments in #726 - this forces parent/child propagation semantics for background jobs. Should we use links instead, and also include the option for the propagation kind an end-user wants?

Reply:
You still need to inject for links, but otherwise I agree.